frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -83,19 +121,27 @@ const setup = async ({
   const mockEventListener = jest.spyOn(window, "addEventListener");
 
   renderWithProviders(
-    <Route path="/question">
+    <Route>
       <IndexRoute component={TestQueryBuilder} />
-      <Route path="notebook" component={TestQueryBuilder} />
-      <Route path=":slug" component={TestQueryBuilder} />
-      <Route path=":slug/notebook" component={TestQueryBuilder} />
+      <Route path="/model">
+        <Route path=":slug/query" component={TestQueryBuilder} />
+        <Route path=":slug/metadata" component={TestQueryBuilder} />
+      </Route>
+      <Route path="/question">
+        <Route path="notebook" component={TestQueryBuilder} />
+        <Route path=":slug" component={TestQueryBuilder} />
+        <Route path=":slug/notebook" component={TestQueryBuilder} />
+      </Route>
     </Route>,
     {
       withRouter: true,
       initialRoute,
     },
   );
 
-  await waitForElementToBeRemoved(() => screen.queryByText(/Loading/));
+  await waitForElementToBeRemoved(() =>
+    screen.queryByTestId("loading-spinner"),
```

Comment:
afaik `queryBy` is usually used when you want *not* to find an element (to check that it doesn't present)

Reply:
Yep - that was the goal for this at least, I wanted to follow what was here in the Notion page (https://www.notion.so/metabase/Mocking-entity-API-endpoints-b3c2cc113c9844a0822783d05ca2d508?pvs=4) but make sure we use the `testid` for some level of consistency with the tests I've seen

Reply:
sorry, my comment wasn't clear :) I proposed to use `getByTestId` instead of `queryByTestId`, as `waitForElementToBeRemoved` adds a wrapper to handle falsy case anyway. But I see in code we use queryBy* with waitForElementToBeRemoved, so I'm fine with it

Reply:
seems, even docs show `queryBy` in this case, so sorry for noise :)